cloud/scope/common.go

```diff
@@ -14,7 +14,9 @@ import (
 	"github.com/linode/cluster-api-provider-linode/version"
 )
 
-func CreateLinodeClient(apiKey string) (*linodego.Client, error) {
+type clientOption func(*linodego.Client)
+
+func CreateLinodeClient(apiKey string, opts ...clientOption) (*linodego.Client, error) {
```

Comment:
I'm not a huge fan of the way `clientOption` isn't exported despite it being used and present in the package's exported API. This change also seems disproportionate to the (presumably-intended) effect.

In the one case in this PR where an option is provided &mdash; `cloud/scope/machine.go:85-87` &mdash; it is being used to disable retries by calling `(*linode.Client).SetRetryCount(0)`. Can that lone call be moved into `CreateLinodeClient` (thus disabling retries for all clients instantiated through this function), or can it be directly called on the `*linode.Client` returned from this function in the one place it's needed?

Reply:
I've observed several other failing E2E suites, but haven't looked into them enough to know the impact. It definitely does seem that linodemachine controller relies on retries the most.